populus/project.py

```diff
@@ -159,46 +162,60 @@ def user_config(self):
 
     @property
     def project_config(self):
-
         if self._project_config_cache is None:
             project_config = Config(
                 config=self._project_config,
-                schema=self._project_config_schema,
             )
             project_config.unref()
             self._project_config_cache = project_config
 
+            # schema validation
+            # partial project config must be merged to get the entire schema
+            if self._user_config_cache is None:
+                self.user_config
+            self._merge_configs_and_validate_schema()
+
         return self._project_config_cache
 
+    def _merge_configs_and_validate_schema(self):
+        if self._merged_config_cache is None:
+            merged_config = copy.deepcopy(self._user_config_cache)
+            for key, value in self._project_config_cache.items(flatten=True):
+                if key != 'version':
+                    merged_config[key] = value
+            self._merged_config_cache = merged_config
+            # schema validation
+            Config(config=merged_config, schema=self._config_schema)
```

Comment:
This doesn't feel like the right way to perform validation.  Performing validation in this manner is indirect and fragile.  Can you modify this so that the validation is being done directly.